src/components/_navigation/TopbarViewer/TopbarViewer.tsx

```diff
@@ -81,21 +87,26 @@ export const TopbarViewer: React.FC = () => {
         </CSSTransition>
       </SearchbarContainer>
       <ButtonWrapper>
-        {mdMatch && (
-          <Button
-            to={absoluteRoutes.studio.index()}
-            newTab
-            icon={<SvgActionAddVideo />}
-            iconPlacement="left"
-            size="medium"
-          >
-            Start publishing
-          </Button>
-        )}
-        {!searchQuery && !mdMatch && (
-          <StyledIconButton to={absoluteRoutes.studio.index()} newTab>
-            <SvgActionAddVideo />
-          </StyledIconButton>
+        {mdMatch &&
+          (isLoggedIn ? (
+            <Button
+              icon={<SvgActionAddVideo />}
+              iconPlacement="left"
+              size="medium"
+              onClick={handleLogging}
+              variant="secondary"
+            >
+              Upload video
+            </Button>
+          ) : (
+            <Button icon={<SvgMember />} iconPlacement="left" size="medium" onClick={handleLogging}>
+              Sign up
+            </Button>
+          ))}
+        {!searchQuery && !mdMatch && !isLoggedIn && (
+          <>
+            <StyledIconButton onClick={handleLogging}>Sign up</StyledIconButton>
+          </>
```

Comment:
```suggestion
          <StyledIconButton onClick={handleLogging}>Sign up</StyledIconButton>
```
You don't need fragment here